whois-query/src/main/java/net/ripe/db/whois/query/planner/AbuseCFinder.java

```diff
@@ -46,61 +47,97 @@ public AbuseCFinder(final RpslObjectDao objectDao,
         this.abuseValidationStatusDao = abuseValidationStatusDao;
     }
 
-    public Optional<AbuseContact> getAbuseContact(final RpslObject object) {
-        final RpslObject role = getAbuseContactRole(object);
-        return role != null?
-                Optional.of(new AbuseContact(
+    public Optional<AbuseContact> getAbuseContact(final RpslObject rpslObject) {
+        final RpslObject role = getAbuseContactRole(rpslObject);
+        if (role == null) {
+            return Optional.empty();
+        }
+
+        final boolean suspect = abuseValidationStatusDao.isSuspect(role.getValueForAttribute(AttributeType.ABUSE_MAILBOX));
+
+        return Optional.of(new AbuseContact(
                         role.getKey(),
                         role.getValueForAttribute(AttributeType.ABUSE_MAILBOX),
-                        abuseValidationStatusDao.isSuspect(role.getValueForAttribute(AttributeType.ABUSE_MAILBOX)),
-                        getOrgToContact(object)
-                )) : Optional.empty();
+                        suspect,
+                        getOrgToContact(rpslObject, suspect)
+                ));
     }
 
     @Nullable
-    private CIString getOrgToContact(final RpslObject object) {
-        if (object.containsAttribute(AttributeType.SPONSORING_ORG)) {
-            return object.getValueForAttribute(AttributeType.SPONSORING_ORG);
+    private CIString getOrgToContact(final RpslObject rpslObject, final boolean suspect) {
+        if (suspect) {
+            final CIString lir = findResponsibleLirOrgReference(rpslObject);
+            if (lir != null) {
+                return lir;
+            }
         }
 
-        final RpslObject responsibleOrg = findResponsibleOrgObject(object);
-        return responsibleOrg != null? responsibleOrg.getValueForAttribute(AttributeType.ORG) : null;
+        return findResponsibleOrgReference(rpslObject);
     }
 
     @Nullable
-    private RpslObject findResponsibleOrgObject(final RpslObject rpslObject) {
+    private CIString findResponsibleOrgReference(final RpslObject rpslObject) {
```

Comment:
findResponsibleOrgReference() and findResponsibleLirOrgReference() are practically identical apart from findResponsibleLirOrgReference() checking the org is an lir, maybe have just one method which also takes on the org type we're looking for as an argument so we don't duplicate the logic..

Reply:
Agreed, I refactored the methods